.github/workflows/test_and_build.yml

```diff
@@ -0,0 +1,81 @@
+name: test_and_build
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.5, 3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install pipenv
+      run: pip3 install pipenv
+
+    - name: Install core dependencies
+      run: pipenv run pip install -r core-requirements.txt
+
+    - name: Test core
+      env:
+          COGNITE_API_KEY: ${{ secrets.COGNITE_API_KEY }}
+          COGNITE_PROJECT: python-sdk-test
+          COGNITE_BASE_URL: https://greenfield.cognitedata.com
+          COGNITE_CLIENT_NAME: python-sdk-integration-tests
+          CI: 1
+      run: pipenv run pytest tests/tests_unit -m 'not dsl' --test-deps-only-core
+
+    - name: Install all dependencies
+      if: matrix.python-version != 3.5
+      run: |
+        pipenv sync --dev
+        sudo npm install -g swagger-cli
+
+    - name: Check codestyle
+      if: matrix.python-version != 3.5
+      run: pipenv run black -l 120 --check .
+
+    - name: Test OpenAPI Generator
+      if: matrix.python-version != 3.5
+      run: pipenv run pytest openapi/tests
+
+    - name: Run tests
+      if: matrix.python-version != 3.5
```

Comment:
So we don't test 3.5?

Could we have an extra job only running a compatible Python version which installs dev dependencies where we check the code and all that. And then only install prod-dependencies + what is necessary for running tests in this python-version matrix?

Reply:
we test 3.5 core only

Reply:
I will see what I can do